config/authorino/rbac/role.yaml

```diff
@@ -0,0 +1,45 @@
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  creationTimestamp: null
+  name: manager-role
+rules:
+- apiGroups:
+  - authorino.3scale.net
+  resources:
+  - authconfigs
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - authorino.3scale.net
+  resources:
+  - authconfigs/status
+  verbs:
+  - get
+  - patch
+  - update
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - create
+  - get
+  - list
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
```

Comment:
We should change in Authorino first, but cannot forget reflecting the change here as well... We're missing a couple rules as part of this role:

```yaml
- apiGroups: ["authentication.k8s.io"]
  resources:
  - tokenreviews
  verbs: ["create"]
- apiGroups: ["authorization.k8s.io"]
  resources:
  - subjectaccessreviews
  verbs: ["create"]
```

These rules have never been missed really only because we are always deploying the `kube-rbac-proxy` sidecar and with that all the RBAC that provides Authorino's SA with exactly the same permission. However, one thing should not rely on the other – `kube-rbac-proxy` creates `tokenreviews` and `subjectaccessreviews` to protect the `/metrics` endpoint; Authorino creates `tokenreviews` and `subjectaccessreviews` as part of the implementation of the [Kubernetes authn](https://github.com/Kuadrant/authorino/blob/main/docs/architecture.md#kubernetes-authentication) and [Kubernetes authz](https://github.com/Kuadrant/authorino/blob/main/docs/architecture.md#kubernetes-authorization) features.

Reply:
Right, I added this permission to the ClusterRole.

@guicassolato When the operator is ready for the first release should we remove those RBAC rules from the authorino repo and have it only on the repo of the operator?

Reply:
> Right, I added this permission to the ClusterRole.
> 
> @guicassolato When the operator is ready for the first release should we remove those RBAC rules from the authorino repo and have it only on the repo of the operator?

Actually from @eguzki [comment](https://github.com/Kuadrant/authorino-operator/pull/1#discussion_r742068502), we can generate it automatically from authorino's repo so there is no need for the operator to host the CRDs and permissions.